src/daos/current-year-dao.js

```diff
@@ -0,0 +1,97 @@
+//For this function we need to first update ydsd table so that gradeEntering is incremented and year is incremented, and then we can increment the year. Consider changing method name for this reason.
+export function incrementDashYear(db) {
+  return new Promise((resolve, reject) => {
+    getDashYear(db)
```

Comment:
You don't need a separate query to get the current dash year, just use a second SELECT statement in the query